### ISSUE TITLE
Anyone can upgrade & be promoted to become a Billing Leader

### ISSUE THREAD

**Issue by jordanh** [ParabolInc/action#3100]
## Issue - Enhancement

When I want to upgrade Parabol to a Pro account, I want to be able to put in my credit card details (vs. having to contact a billing leader), so that I can pay.

When I put in my credit card details, I want to become the billing leader, so that I can edit my payment details later.

(also: the fact that I possess a credit card capable of paying _makes_ me _de facto_ the billing leader)

### Acceptance Criteria (optional)
Users can:
   - Raise the payment modal, enter payment details
   - If the payment is valid, promote them to billing leader as they are upgrade to Pro

- **Estimated effort:** 12 points ([see CONTRIBUTING.md](https://github.com/ParabolInc/action/blob/master/CONTRIBUTING.md#points-and-sizes))

**Comment by mattkrick** [comment]
Stuck on:
- what if i update the already existing payment details? does that make me a leader? if so, that means anyone with a CC can take over the org. if not & the billing leader leaves the org, then the billing leader tag must be shared prior to them leaving or we'll have a support ticket on our hands.
- if you're the billing leader & i enter my CC number, then you have the power to unmark me as billing leader. if that happens, how do i remove my CC?

**Comment by ackernaut** [comment]
- I think that it’s safe to try for now, that anybody can update CC info and becomes a billing leader
- Only billing leaders can downgrade
- We track which billing leader last updated the CC info. They are the default billing leader. Another billing leader can demote a default billing leader only if they update with new CC info.

**Comment by mattkrick** [comment]
Current billing leader privileges
- upgrade (add a CC)
- downgrade (remove a CC)
- update a CC
- remove org member
- change org name/avatar
- view invoice
- promote org member to billing leader
- remove org member from billing leader
- delete the org (future)

Before we start rackin up a bunch of tech debt, here's what sentry & stripe have:

![Screenshot from 2019-08-15 18-40-37](https://user-images.githubusercontent.com/5514175/63137766-58f09800-bf8c-11e9-8bc0-8c9b2e1495d8.png)

![Screenshot from 2019-08-15 18-43-53](https://user-images.githubusercontent.com/5514175/63137851-b1279a00-bf8c-11e9-8abc-4c1038f5a3b8.png)

- if we think of free users as trial users, none of it matters, so i'm happy to let anyone upgrade.
- downgrading affects billing, so you should have to be a billing leader to downgrade
- updating an invalid CC is the same as a free user, so anyone can do it
- updating a _valid_ CC is tricky. the disgruntled ex-employee shouldn't be able to ruin an entire org by becoming a billing leader, deleting everyone, then deleting the org for a full refund. that's a lot of destruction for free. so, ***updating valid CC must require the billing leader permission***
- removing an org member should be possible without being a billing leader. i'd argue this should be the role of the team leader. maybe in the future if you remove someone from the last team of that org, they're gone from the org.
- change org name/avatar has nothing to do with billing
- view invoice probably makes sense for the billing leader, although it's read only, so if there were a role below billing leader, that'd be OK too
- promote to billing leader- must be billing leader
- remove from billing leader- IIRC reddit using a ranking system (newer mods cannot remove older mods) i think GH is the same. 
- delete the org. honestly, billing leader may not be strong enough for this. only 1 person should be able to do this one.

I think the single `billing leader` role is still sufficient, although it'd be nice to have an OWNER, BILLING LEADER(s), and MANAGER(s).  for now, i think team leads double as managers and making team-of-teams will help solidify that. an owner would be nice to keep 1 billing leader from going rogue, but not necessary i suppose.
